src/App.js

```diff
@@ -1,43 +1,87 @@
 import React from "react";
 import Joke from "./components/Joke";
 import PageHeader from "./components/PageHeader";
+import SearchInput from "./components/SearchInput";
+import PropTypes from "prop-types";
 
 class App extends React.Component {
-  constructor() {
-    super()
-    this.state = {
-      jokeData: {
-        results: []
-      }
+  state = {
+    jokeData: {
+      results: []
+    },
+    searchTerm: "",
+    loadingState: "resting"
+  };
+
+  handleChange = event => {
+    this.setState({ searchTerm: event.target.value });
+  };
+
+  handleClick = event => {
+    event.preventDefault();
+    this.setState({ loadingState: "loading" });
+  };
+
+  async fetchJokes() {
+    try {
+      const url =
+        this.state.searchTerm === ""
+          ? `https://icanhazdadjoke.com/search`
+          : `https://icanhazdadjoke.com/search?term=${this.state.searchTerm}`;
+      const fetchUrl = fetch(url, {
+        method: "GET",
+        headers: { accept: "application/json" }
+      });
+      const response = await fetchUrl;
+      const myData = await response.json();
+      this.setState({
+        jokeData: { results: myData.results },
+        loadingState: "loaded"
+      });
+    } catch (e) {
+      this.setState({ loadingState: "Something went wrong" });
     }
   }
 
-  async componentDidMount() {
-    const url = 'https://icanhazdadjoke.com/search?term="cat"';
-    const fetchUrl = fetch(url, {
-      method: "GET",
-      headers: { accept: "application/json" }
-    });
-    const response = await fetchUrl;
-    const myData = await response.json();
-    this.setState({ jokeData: { results: myData.results } });
+  componentDidUpdate(prevProps, prevState) {
```

Comment:
Any time I see `componentDidUpdate` being used it makes me double check... you sure you need this? What problem are you trying to solve by adding this?

Reply:
In the handleSubmit, the state is set to loading when the user submits the search. In the `componentDidUpdate()` if Loading state = loading, then it fetches those jokes with necessary term.